lib/standalone.mjs

```diff
@@ -292,6 +304,58 @@ export default async function standalone({
 		stylelintResults = await Promise.reject(new NoFilesFoundError(fileList));
 	}
 
+	/** @type {Record<string, unknown>} */
+	let unusedSuppressions = {};
+
+	if (!useInputCode) {
+		const resolvedSuppressLocation = resolveFilePath(
+			suppressLocation || DEFAULT_SUPPRESSION_FILENAME,
+			cwd,
+			DEFAULT_SUPPRESSION_FILENAME,
+		);
+
+		const existsSuppressionsFile = existsSync(resolvedSuppressLocation);
+
+		if (suppressLocation && !existsSuppressionsFile && !suppressAll && !suppressRule) {
+			throw new SuppressionFileNotFoundError();
+		}
+
+		if (suppressAll || suppressRule || suppressPrune || existsSuppressionsFile) {
+			emitExperimentalWarning(
+				'The suppressions feature is experimental.',
+				'SUPPRESSIONS',
+				'See https://stylelint.io/user-guide/suppressions for more information.',
+			);
+
+			const suppressions = new SuppressionsService({
+				filePath: resolvedSuppressLocation,
+				cwd: process.cwd(),
+			});
+
+			if (suppressAll || suppressRule) {
+				await suppressions.suppress(stylelintResults, suppressRule);
+			}
+
+			if (suppressPrune) {
+				await suppressions.prune(stylelintResults);
+			}
+
+			const suppressionResults = suppressions.applySuppressions(
+				stylelintResults,
+				await suppressions.load(),
+			);
+
+			stylelintResults = suppressionResults.results;
+			unusedSuppressions = suppressionResults.unused;
+		}
+	}
+
+	const unusedSuppressionsCount = Object.keys(unusedSuppressions).length;
+
+	if (unusedSuppressionsCount > 0 && !suppressPrune) {
```

Comment:
[suggestion] Minor. I'm not 100% sure, but by evaluating `suppressPrune` first, we might save a bit of the calculation and memory cost for `Object.keys()`:

```suggestion
	if (!suppressPrune && Object.keys(unusedSuppressions).length > 0) {
```